Kernel/Syscall.cpp

```diff
@@ -285,12 +285,7 @@ static u32 handle(RegisterDump& regs, u32 function, u32 arg1, u32 arg2, u32 arg3
     case Syscall::SC_sched_getparam:
         return current->process().sys$sched_setparam((pid_t)arg1, (struct sched_param*)arg2);
     case Syscall::SC_halt: {
-        dbgprintf("<%u> halting! acquiring locks...\n");
-        FS::lock_all();
-        dbgprintf("<%u> halting! syncing...\n");
-        FS::sync();
-        dbgprintf("<%u> halting! bye, friends...\n");
-        IO::out16(0x604, 0x2000);
+        return current->process().sys$halt();
```

Comment:
This looks cleaner, thanks :)

I didn't put it in Process before because that didn't seem clean (it's halting the system, not the process). But of course, syscalls come *from* a process, so this is a good cleanup! :+1:

Reply:
Yeah that's the same reason I put `reboot` in there too, though I suppose it's a Unix'y thing? I also suppose the actual program (shutdown/reboot) shouldn't make a direct call to `syscall`.